addons/mail/static/src/models/thread/thread.js

```diff
@@ -788,7 +788,14 @@ function factory(dependencies) {
          */
         _computeAllAttachments() {
             const allAttachments = [...new Set(this.originThreadAttachments.concat(this.attachments))]
-                .sort((a1, a2) => a1.id < a2.id ? 1 : -1);
+                  .sort((a1, a2) => {
+                      // Handle the temporary attachments that should be at the
+                      // start of the list. Temporary attachments have a negative id.
+                      if (a1.id > 0 && a2.id < 0) {
+                          return 1;
+                      }
+                      return a1.id < a2.id ? 1 : -1;
+                  });
```

Comment:
I'll test to better understand what is going on, but I can already tell the indent here should be 4 not 6, and also the PR is missing a test assertion this new order.

Reply:
This comment was not done btw, indent still wrong and missing a test, will make new task.